src/components/ArticlesList.jsx

```diff
@@ -19,6 +25,10 @@ const ArticleList = () => {
       });
   }, []);
 
+  const filteredArticles = selectedTopic
+  ? articles.filter((article) => article.topic === selectedTopic)
```

Comment:
Instead of filtering by topic here, please fetch articles by topic from your API using 'axios params'.